pallets/api/src/messaging/mod.rs

```diff
@@ -761,48 +846,78 @@ impl<T: Config> crate::Read for Pallet<T> {
 	}
 }
 
+/// Represents a cross-chain message in the system.
+///
+/// Each variant of this enum captures a different state or type of message lifecycle:
+/// - A request in progress.
+/// - A response received.
+/// - A timeout occurred.
+///
+/// This is used internally to track, manage, and clean up messages, along with any
+/// associated deposits and optional callback metadata.
 #[derive(Clone, Debug, Encode, Eq, Decode, MaxEncodedLen, PartialEq, TypeInfo)]
 #[scale_info(skip_type_params(T))]
 pub enum Message<T: Config> {
```

Comment:
if this is used internally, would we be able to use a `pub(crate)` instead, or should it still be pub?